src/spend.rs

```diff
@@ -172,6 +172,29 @@ pub struct CandidateCoin {
     pub sequence: Option<bitcoin::Sequence>,
 }
 
+/// A coin selection result.
+///
+/// A change output should only be added if `change_amount >= change_policy_min_amount`.
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct CoinSelectionRes {
+    /// Selected candidates.
+    pub selected: Vec<CandidateCoin>,
+    /// Amount the selected candidates overpay the amount required to pay all recipients
+    /// at the target feerate, taking into account the change output, if any, as determined
+    /// by the change policy.
+    ///
+    /// A non-zero value implies that either a possible change output has not been added
+    /// by the change policy or otherwise that the excess amount is too small to pay
+    /// for a change output.
+    pub excess: bitcoin::Amount,
+    /// Change amount that could be included for this selection while still respecting the
+    /// target feerate, irrespective of the change policy used for selection.
+    pub change_amount: bitcoin::Amount,
+    /// Minimum change amount required in order for a change output to be added
+    /// to the transaction, as determined by the change policy used for this selection.
+    pub change_policy_min_amount: bitcoin::Amount,
```

Comment:
Why can't we just have:
- `excess`: the amount overpaid, with or without change
- `change_amount`: the change amount, if non-zero

It seems to me that's all we need? If `change_amount` is 0 and `excess` is >0 (or to a given threshold?) emit a warning.

Reply:
Hmm yes I think we could just include the `excess` taking into account any change added according to the policy and keep the `change_amount` as the amount of change according to the policy, and remove `change_policy_min_amount` (the caller could then continue to use `change_amount` directly without needing to check the threshold). This should be clearer as then all values would be according to the change policy.

Then the `excess` would be the amount being overpaid with respect to the target feerate, which is what we want. The possible amount of change in the absence of any min threshold would be smaller than the excess as we'd need to pay for the change output, but I think the important thing is to know how much is being overpaid rather than how much change there might have been if a change output had been included.

Reply:
This has now been done.